lib/scope.js

```diff
@@ -805,8 +813,80 @@ module.exports = {
 
   getMatchingRow: async function ( scope, { story_table, page_data }) {
     /* Given html_data and a list of objects describing fields and values,
-    returns the matching fields and the data necessary to set those values. */
-    return scope;
+    * returns the matching fields and the data necessary to set those values.
+    * May also handle proxy vars (e.g. x, i...)
+    * 
+    * @param story_table {arr} - fields and the values they can be set to
+    * @param story_table[n].var_name {str} - name of variable to set
+    * @param story_table[n].value {str} - choice to set or value to set
+    * @param story_table[n].checked {str} - if a choice, whether to select
+    *    select it or un-select it
+    * 
+    * @param page_data {obj} - contains page data
+    * @param page_data.sought_var {arr} - list (why?) containing the actual
+    *    string name of the var that triggered the question. Should contain
+    *    info that can help us tease out proxy variable parts (e.g. x, i...)
+    * @param page_data.fields {arr} - array of data for each field on the page
+    * @param page_data.fields[n].selector {str} - used to find the DOM node
+    * @param page_data.fields[n].tag {str} - HTML tag of node
+    * @param page_data.fields[n].type {str} - HTML type of node if it has one
+    * @param page_data.fields[n].rows {arr} - arr of objects that could
+    *    possibly match a table row
+    * @param page_data.fields[n].rows[n].var_name {str} - name of variable
+    * @param page_data.fields[n].rows[n].value {str} - value of a choice
+    * @param page_data.fields[n].rows[n].checked {str} - '' if the node
+    *    cannot be checked, otherwise `true` if it is checked and `false` otherwise
+    */
+    // There should only ever be one match for any table row and for any field row.
+    // Not sure how to check on this.
+    let matches = [];
+
+    for ( let field of page_data.fields ) {
+      for ( let guess of field.rows ) {  // usually only one or two
+        for ( let table_row of story_table ) {
+          // Check different things depending on the type of field
+          // canvas, buttons, checkboxes, radios, dropdowns, inputs, textareas
+          let rows_match = false;
+          let names_match = table_row.var_name === guess.var_name;
+          let names_and_values_match = names_match && table_row.value === guess.value;
+
+          switch ( field.tag ) {
+            case 'canvas':
+              if ( names_and_values_match ) { rows_match = true; }
+              break;
+
+            case 'button':
+              if ( names_and_values_match ) { rows_match = true; }
+              break;
```

Comment:
Since button and canvas have the same code, you could combine them

```javascript
case 'canvas': 
case 'button': 
    # something 
```

Reply:
True! Making into an issue.